### PR TITLE
add SpecifiedByUrl for custom Scalars

### PR DESCRIPTION
**Pull request by no1seman**
add SpecifiedByUrl for custom Scalars and also SpecifiedByUrl directive acc. #3.5, #4.2.2 of GraphQL spec draft: http://spec.graphql.org/draft/

### CODE REVIEW

File: test/integration/graphql_test.lua [hunk 1]
```diff
@@ -1371,6 +1371,9 @@ function g.test_specifiedByUrl()
     }
 
     local data, errors = check_request(introspection.query, query_schema)
-    t.assert_equals(tostring(util.map_name(data.__schema.types, function(v) return v end)['CustomInt'].specifiedByUrl), 'http://localhost')
+    t.assert_equals(
+        tostring(util.map_name(data.__schema.types, function(v) return v end)['CustomInt'].specifiedByUrl),
+        'http://localhost'
+    )
```
Comment (olegrok) [inline test/integration/graphql_test.lua]: This commit should be squashed into previous one.

File: test/integration/graphql_test.lua [hunk 2]
```diff
@@ -1337,3 +1338,39 @@ function g.test_custom_directives()
...
+                arg = custom_scalar,
+            },
+            resolve = callback,
+        }
+    }
+
+    local data, errors = check_request(introspection.query, query_schema)
+    t.assert_equals(tostring(util.map_name(data.__schema.types, function(v) return v end)['CustomInt'].specifiedByUrl), 'http://localhost')
+    t.assert_equals(errors, nil)
+end
```
Comment (olegrok) [inline test/integration/graphql_test.lua:1379]: Git warns about missing empty line

File: test/integration/graphql_test.lua [hunk 3]
```diff
@@ -1337,3 +1338,39 @@ function g.test_custom_directives()
...
+            kind = types.string.nonNull,
+            arguments = {
+                arg = custom_scalar,
+            },
+            resolve = callback,
+        }
+    }
+
+    local data, errors = check_request(introspection.query, query_schema)
+    t.assert_equals(tostring(util.map_name(data.__schema.types, function(v) return v end)['CustomInt'].specifiedByUrl), 'http://localhost')
```
Comment (olegrok) [inline test/integration/graphql_test.lua]: Do you really need tostring here?

**Review by olegrok — COMMENTED** [review]
Commit message still is empty. Please fix it. Also see my comments below.

File: test/integration/graphql_test.lua (hunk 1 above)
Comment (no1seman) [reply test/integration/graphql_test.lua]: ?

Comment (olegrok) [reply test/integration/graphql_test.lua]: [This](https://github.com/tarantool/graphql/pull/21/commits/55e5d41ea7e160368388af5ce5f806c1c1c2ce4a) commit should be squashed into [previous one](https://github.com/tarantool/graphql/pull/21/commits/93a031ffdb05588c15f522c1d6eda31bc4fa48da).

Comment (no1seman) [reply test/integration/graphql_test.lua]: Why? It'a a test for SpecifiedByUrl field for custom Scalar. Please explain more detailed why this have to squashed into previous commit?

Comment (olegrok) [reply test/integration/graphql_test.lua]: I talk about commit "Fix lint warning".
All changes should be atomic. But here you introduces "Add Scalar specifiedByUrl field and specifiedByUrl directive" a code that breaks linter and fixes it the following commit.

"Add Scalar specifiedByUrl field and specifiedByUrl directive" should be self-sufficient and shouldn't request some follow-ups to fix it.

Comment (no1seman) [reply test/integration/graphql_test.lua]: Ok

**Comment by DifferentialOrange** [comment]
I was finishing this PR as a part of this sprint tasks. You can see reworked version in #27 . Since this PR is based on branch of no1seman/graphql and I don't think it's nice to force-push to other people forks' branches, I opened a new one.

**Comment by Totktonada** [comment]
Pushed in PR #27.

**Pull request closed without merging**